### ISSUE TITLE
[Bug]: v0.6.4 requires more GPU memory than v0.6.3

### ISSUE THREAD

**Issue by paulcx** [vllm-project/vllm#10360]
### Your current environment

docker image 0.6.4

### 🐛 Describe the bug

docker run 0.6.4 image requires more GPU memory than version 0.6.3 to trigger OOM with the same settings. The model being used is Qwen 2.5 1.5B, with the following Docker launch settings: --model Qwen2.5-1.5B-Instruct --max-model-len 4096 --gpu-memory-utilization 0.15 --enable-prefix-caching

**Comment by DarkLight1337** [comment]
We have updated our memory profiling to be more accurate in #9352, so the previous memory allocation might have been an underestimation in your case.

**Comment by paulcx** [comment]
ok, it is close for now.

**Comment by paulcx** [comment]
@DarkLight1337 During several weeks of testing, I still found an OOM situation. Here are the parameters and logs to reproduce it

IMAGE: `0.6.4.post1`
GPU: `A6000 (48G)`
CMD: `'--model' 'Qwen2.5-32B-Instruct-AWQ' '--max-model-len' '4096' '--enable-prefix-caching'`
LOG: `Memory profiling results: total_gpu_memory=47.54GiB initial_memory_usage=18.60GiB peak_torch_memory=19.60GiB memory_usage_post_profile=18.62GiB non_torch_memory=0.44GiB kv_cache_size=22.74GiB gpu_memory_utilization=0.90`

**Comment by DarkLight1337** [comment]
cc @joerunde @tlrmchlsmth

**Comment by paulcx** [comment]
> @paulcx can you try again with v0.6.5?

Testing in progress, I will test for a week and then feedback the situation after this update

**Comment by paulcx** [comment]
@DarkLight1337 @joerunde some issues found in v0.6.5

IMAGE: v0.6.5
GPU: A6000 (48G) * 2
CMD: --model Qwen2-VL-72B-Instruct-AWQ --tensor-parallel-size 2 --max-model-len 8000 --limit-mm-per-prompt image=4
LOG: 
```
the current vLLM instance can use total_gpu_memory (47.54GiB) x gpu_memory_utilization (0.90) = 42.78GiB
model weights take 20.05GiB; non_torch_memory takes 0.49GiB; PyTorch activation peak memory takes 1.58GiB; the rest of the memory reserved for KV Cache is 20.66GiB.
Memory profiling takes 11.56 seconds
# GPU blocks: 8450, # CPU blocks: 1638
Maximum concurrency for 8000 tokens per request: 16.90x
```
ERR: ``` CUDA out of memory. Tried to allocate 20.00 MiB. GPU 0 has a total capacity of 47.54 GiB of which 13.75 MiB is free. Process 1130970 has 47.51 GiB memory in use. Of the allocated memory 40.80 GiB is allocated by PyTorch, with 106.00 MiB allocated in private pools (e.g., CUDA Graphs), and 267.84 MiB is reserved by PyTorch but unallocated.```

Somehow it works in v0.6.4.post1 with same config and model:
```
Loading model weights took 20.0492 GB
Loading model weights took 20.0492 GB
Memory profiling results: total_gpu_memory=47.54GiB initial_memory_usage=20.67GiB peak_torch_memory=21.72GiB memory_usage_post_profile=20.81GiB non_torch_memory=0.74GiB kv_cache_size=20.32GiB gpu_memory_utilization=0.90
INFO 12-20 09:54:13 worker.py:232] Memory profiling results: total_gpu_memory=47.54GiB initial_memory_usage=20.67GiB peak_torch_memory=21.72GiB memory_usage_post_profile=20.84GiB non_torch_memory=0.77GiB kv_cache_size=20.29GiB gpu_memory_utilization=0.90
# GPU blocks: 8309, # CPU blocks: 1638
Maximum concurrency for 8000 tokens per request: 16.62x
```

**Comment by joerunde** [comment]
@paulcx does this OOM error happen while vllm is starting up, or when sending requests?

**Comment by paulcx** [comment]
> @paulcx does this OOM error happen while vllm is starting up, or when sending requests?

it happens when sending the first request

**Comment by paulcx** [comment]
> 🤔 That's interesting, can you share the request?

here's my request
```json
{
  "model": "Qwen2-VL-72B-Instruct-AWQ",
  "messages": [
    {
      "role": "user",
      "content": "who are you?"
    }
  ],
  "max_tokens": null,
  "temperature": 0,
  "seed": 42
}
```

**Comment by joerunde** [comment]
Oh that's not good, no images or long prompt? I can't reproduce that crash on a single 80GB A100, it seems to work fine 😞.

Does reducing the kv cache allocation a bit with `--gpu-memory-utilization=0.8` help at all?

**Comment by paulcx** [comment]
> Oh that's not good, no images or long prompt? I can't reproduce that crash on a single 80GB A100, it seems to work fine 😞.
> 
> Does reducing the kv cache allocation a bit with `--gpu-memory-utilization=0.8` help at all?

I run the test on two A6000 (48G). As I mentioned earlier, the memory calculation difference is very small, or there are other changes I don't know about. In summary, using the default gpu-memory-utilization=0.9 causes an out-of-memory error, but 0.85 works fine. My question is why 0.6.4.post1 uses the default 0.9 to work?

**Comment by joerunde** [comment]
Yeah, I think you're right that the memory profiling differences are very small, so I'd agree that some other changes in v0.6.5 are probably using more memory. You can try vLLM's runtime profiling to see what's going on: https://docs.vllm.ai/en/latest/contributing/profiling/profiling_index.html#profiling-vllm
